src/dodal/devices/electron_analyser/abstract/base_driver_io.py

```diff
@@ -127,39 +127,17 @@ def __init__(
 
         super().__init__(prefix=prefix, name=name)
 
+    @abstractmethod
     @AsyncStatus.wrap
     async def set(self, region: TAbstractBaseRegion):
```

Comment:
```suggestion
    @abstractmethod
    async def set(self, region: TAbstractBaseRegion) -> AsyncStatus:
```

I think right now you're wrapping the method that returns an AsyncStatus in an AsyncStatus and the move would appear to be done immediately